libs/hdf_converters/src/aws-config-mapper.ts

```diff
@@ -0,0 +1,345 @@
+    if (response === undefined) {
+      throw new Error('No data was returned');
+    } else {
+      while (
+        response !== undefined &&
+        response.ConfigRules !== undefined &&
+        response.NextToken !== undefined
+      ) {
+        response.ConfigRules.forEach((rule) => {
+          configRules.push(rule);
+        });
+        params = _.set(params, 'NextToken', response.NextToken);
+        response = await this.getConfigRulePage(params);
+      }
+    }
+    return this.addResultsToConfigRules(
+      this.addComplianceToConfigRules(configRules)
+    );
+  }
+  private async getConfigRulePage(
+    params: DescribeConfigRulesCommandInput
+  ): Promise<DescribeConfigRulesCommandOutput> {
+    return this.configService.describeConfigRules(params);
+  }
+  private addResultsToConfigRules(configRules: ConfigRule[]): ConfigRule[] {
+    configRules.forEach(async (rule) => {
+      let params = {
+        ConfigRuleName: rule.ConfigRuleName,
+        Limit: 100
+      };
+      let response = await this.configService.getComplianceDetailsByConfigRule(
+        params
+      );
+      let ruleResults = response.EvaluationResults || [];
+      do {
+        params = _.set(params, 'NextToken', response.NextToken);
+        response = await this.configService.getComplianceDetailsByConfigRule(
+          params
+        );
+        ruleResults = ruleResults?.concat(response.EvaluationResults || []);
+      } while (response.NextToken !== undefined);
+      rule = _.set(rule, 'results', []);
+      ruleResults.forEach((result) => {
+        const hdfResult: ExecJSON.ControlResult = {
+          code_desc: this.getCodeDesc(result),
+          start_time: result.ConfigRuleInvokedTime?.toDateString() || '',
+          run_time: this.getRunTime(result),
+          status: this.getStatus(result),
+          message: this.getMessage(result, this.getStatus(result))
+        };
+        rule = _.set(rule, 'results', _.get(rule, 'results').push(hdfResult));
+      });
+      switch (rule.ConfigRuleName) {
+        case 'NOT_APPLICABLE': {
+          rule = _.set(rule, 'impact', 0);
+          rule = _.set(
+            rule,
+            'results',
+            _.get(rule, 'results').concat([
+              {
+                run_time: 0,
+                code_desc: NOT_APPLICABLE_MSG,
+                skip_message: NOT_APPLICABLE_MSG,
+                start_time: Date.now(), //GOTTA FIGURE THIS OUT TOO
+                status: 'skipped'
+              }
+            ])
+          );
+          break;
+        }
+        case 'INSUFFICIENT_DATA': {
+          rule = _.set(rule, 'impact', 0);
+          rule = _.set(
+            rule,
+            'results',
+            _.get(rule, 'results').concat([
+              {
+                run_time: 0,
+                code_desc: INSUFFICIENT_DATA_MSG,
+                skip_message: INSUFFICIENT_DATA_MSG,
+                start_time: Date.now(), //GOTTA FIGURE THIS OUT TOO
```

Comment:
Is this comment still relevant?

Reply:
Yes, I have to do a bit more reading into the Date module to find out how to convert it into something readable rather than UTC